crates/circuit/src/dag_circuit.rs

```diff
@@ -434,40 +415,64 @@ struct DAGVarInfo {
     out_node: NodeIndex,
 }
 
+#[derive(Clone, Debug)]
+struct VarsByType {
+    input: OnceLock<Py<PySet>>,
+    capture: OnceLock<Py<PySet>>,
+    declare: OnceLock<Py<PySet>>,
+}
```

Comment:
Maybe add a docstring on top of this? I know the type is self-explanatory, but it would be nicely presented with a docstring on top :)

Reply:
Added in: https://github.com/Qiskit/qiskit/pull/14069/commits/78698dd3beecb89a147903817a2bb9f09be6efac